turbopack/crates/turbo-tasks-backend/src/utils/dash_map_multi.rs

```diff
@@ -0,0 +1,178 @@
+use std::{
+    hash::{BuildHasher, Hash},
+    ops::{Deref, DerefMut},
+    sync::Arc,
+};
+
+use dashmap::{DashMap, RwLockWriteGuard, SharedValue};
+use hashbrown::{hash_map, HashMap};
+
+pub enum RefMut<'a, K, V, S>
+where
+    S: BuildHasher,
+{
+    Base(dashmap::mapref::one::RefMut<'a, K, V, S>),
+    Simple {
+        #[allow(dead_code)]
+        guard: RwLockWriteGuard<'a, HashMap<K, SharedValue<V>, S>>,
+        key: *const K,
+        value: *mut V,
+    },
+    Shared {
+        #[allow(dead_code)]
+        guard: Arc<RwLockWriteGuard<'a, HashMap<K, SharedValue<V>, S>>>,
```

Comment:
I think an underscore prefix should silence the lint here.

```suggestion
        _guard: RwLockWriteGuard<'a, HashMap<K, SharedValue<V>, S>>,
        key: *const K,
        value: *mut V,
    },
    Shared {
        _guard: Arc<RwLockWriteGuard<'a, HashMap<K, SharedValue<V>, S>>>,
```